### PR TITLE
[Locator] External services support.

### PR DESCRIPTION
**Pull request by kobolog**
Allow external services to be bound to Locator and propagated throughout the cluster, including IPVS magic and stuff.

### CODE REVIEW

**Comment by kobolog** [comment]
@3Hren: WTB review!

File: src/service/locator.cpp [hunk 1]
```diff
@@ -163,6 +165,61 @@ locator_t::remote_t::on_shutdown() {
...
+    }
+
+private:
+    void
+    discard(const std::error_code& ec, const std::string& handle) {
+        COCAINE_LOG_INFO(parent->m_log, "alien disconnected: [%d] %s", ec.value(), ec.message())(
+            "service", handle
+        );
+
+        return parent->on_service(handle, results::resolve{}, 0);
```
Comment (antmat) [inline src/service/locator.cpp]: 0 for false? why not false itself?

**Comment by antmat** [comment]
So alien services will be accessible only via metalocator. Right?

File: src/service/locator.cpp (hunk 1 above)
Comment (kobolog) [inline src/service/locator.cpp]: It's changed to a human-readable enum in another PR, so ignore this 8)

**Comment by kobolog** [comment]
That's the idea. So that you always go thru IPVS. Might be not the best one, discuss!

**Comment by antmat** [comment]
LGTM

File: src/service/locator.cpp (hunk 1 above)
Comment (kobolog) [inline src/service/locator.cpp]: https://github.com/cocaine/cocaine-core/pull/185/files#diff-a553a12730705141f40f10a7f125a2a4R156

**Pull request merged**